cmd/rookctl/object/connection.go

```diff
@@ -104,14 +104,18 @@ func getConnectionInfo(c client.RookRestClient, storeName, userID, format string
 
 		// write object store connection info
 		fmt.Fprintf(w, PrettyOutputFmt, AWSHost, connInfo.Host)
-		fmt.Fprintf(w, PrettyOutputFmt, AWSEndpoint, connInfo.IPEndpoint)
+		for _, port := range connInfo.Ports {
+			fmt.Fprintf(w, PrettyOutputFmt, AWSEndpoint, fmt.Sprintf("%s:%d", connInfo.IPAddress, port))
+		}
 		fmt.Fprintf(w, PrettyOutputFmt, AWSAccessKey, *user.AccessKey)
 		fmt.Fprintf(w, PrettyOutputFmt, AWSSecretKey, *user.SecretKey)
 
 		w.Flush()
 	} else if format == FormatEnvVar {
 		buffer.WriteString(fmt.Sprintf(ExportOutputFmt, AWSHost, connInfo.Host))
-		buffer.WriteString(fmt.Sprintf(ExportOutputFmt, AWSEndpoint, connInfo.IPEndpoint))
+		for _, port := range connInfo.Ports {
+			buffer.WriteString(fmt.Sprintf(ExportOutputFmt, AWSEndpoint, fmt.Sprintf("%s:%d", connInfo.IPAddress, port)))
```

Comment:
what effect is this going to have on users that do a `eval $(rookctl object connection rook-user --format env-var)` as instructed in https://rook.io/docs/rook/master/client.html#object-storage?

i guess the last one printed will be the one that gets set in their environment, since each iteration of the loop is using the same env var.

i don't think that's a bad thing necessarily, but i just wanted to be clear.

Reply:
right, the last one will win. If they care they'll have to manage it more closely